### PR TITLE
Make PeerManager own a FastRandomContext

### PR DESCRIPTION
**Pull request by dergoegge**
This lets us avoid some non-determinism in tests (also see #28537).

### CODE REVIEW

**Comment by dergoegge** [comment]
cc @MarcoFalke

File: src/test/fuzz/fees.cpp [hunk 1]
```diff
@@ -17,7 +17,8 @@ FUZZ_TARGET(fees)
 {
     FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
     const CFeeRate minimal_incremental_fee{ConsumeMoney(fuzzed_data_provider)};
-    FeeFilterRounder fee_filter_rounder{minimal_incremental_fee};
+    FastRandomContext rng;
```
[the remark this answers is not in the document]
Comment (dergoegge) [reply src/test/fuzz/fees.cpp]: fixed

File: src/net_processing.cpp [hunk 2]
```diff
@@ -695,6 +695,10 @@ class PeerManagerImpl final : public PeerManager
     /** Send `feefilter` message. */
     void MaybeSendFeefilter(CNode& node, Peer& peer, std::chrono::microseconds current_time) EXCLUSIVE_LOCKS_REQUIRED(g_msgproc_mutex);
 
+    FastRandomContext m_rng;
```
[the remark this answers is not in the document]
Comment (dergoegge) [reply src/net_processing.cpp]: Added the annotations.

[the remark this answers is not in the document]
Comment (dergoegge) [reply src/net_processing.cpp]: > Also, could add a test-only method to (re)set the state?

There's a lot of state in peerman that needs resetting and I'd prefer if we'd just make the test setup initialization faster so we can setup a new peerman each iteration.

**Comment by dergoegge** [comment]
Can someone restart the win64 job? failure is unrelated

**Review by maflcko — APPROVED** [review]
lgtm ACK 7022d6a93db08b1334ddae9df439186308bfdd4b 🎇

<details><summary>Show signature</summary>

Signature:

```
untrusted comment: signature from minisign secret key on empty file; verify via: minisign -Vm "${path_to_any_empty_file}" -P RWTRmVTMeKV5noAMqVlsMugDDCyyTSbA3Re5AkUrhvLVln0tSaFWglOw -x "${path_to_this_whole_four_line_signature_blob}"
RUTRmVTMeKV5npGrKx1nqXCw5zeVHdtdYURB/KlyA/LMFgpNCs+SkW9a8N95d+U4AP1RJMi+krxU1A3Yux4bpwZNLvVBKy0wLgM=
trusted comment: lgtm ACK 7022d6a93db08b1334ddae9df439186308bfdd4b 🎇
Vvrl6j+oChrlRcQJFM+xw385dp6qeuK8yw7FZZj2uY0iIWAtdofcaOsmCdXTS0KPV8XWyQzkVw94xJeYmx4GBQ==
```

</details>

File: src/net_processing.cpp [hunk 3]
```diff
@@ -5338,7 +5338,7 @@ void PeerManagerImpl::MaybeSendFeefilter(CNode& pto, Peer& peer, std::chrono::mi
     if (pto.IsBlockOnlyConn()) return;
 
     CAmount currentFilter = m_mempool.GetMinFee().GetFeePerK();
-    static FeeFilterRounder g_filter_rounder{CFeeRate{DEFAULT_MIN_RELAY_TX_FEE}};
+    static FeeFilterRounder g_filter_rounder{CFeeRate{DEFAULT_MIN_RELAY_TX_FEE}, m_rng};
```
[the remark this answers is not in the document]
Comment (dergoegge) [reply src/net_processing.cpp]: Right, I can shuffle the commits to de-globalize before passing in the rng

[the remark this answers is not in the document]
Comment (dergoegge) [reply src/net_processing.cpp]: Done

File: src/test/policy_fee_tests.cpp [hunk 4]
```diff
@@ -13,7 +13,8 @@ BOOST_AUTO_TEST_SUITE(policy_fee_tests)
 
 BOOST_AUTO_TEST_CASE(FeeRounder)
 {
-    FeeFilterRounder fee_rounder{CFeeRate{1000}};
+    FastRandomContext rng;
```
[the remark this answers is not in the document]
Comment (dergoegge) [reply src/test/policy_fee_tests.cpp]: Done

**Review by glozow — COMMENTED** [review]
concept && light code review ACK 4cafe9f176e93ebb6c38abb12140e8d8be005cbf

**Pull request merged**